cheatsheets/Java_Security_Cheat_Sheet.md

```diff
@@ -0,0 +1,971 @@
+
+#### Asymmetric example using Google Tink
+
+Google Tink has documentation on performing common tasks.
+
+For example, this page shows how to perform a hybrid encryption process where two parties want to share data based on their asymmetric key pairs:
+
+[Simple Asymmetric Encryption](https://developers.google.com/tink/encrypt-data) (From Google's website)
+
+The following code snippet shows how this functionality can be used to share secrets between Alice and Bob:
+
+<details>
+  <summary>Click here to expand the code snippet</summary>
+
+``` java
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.crypto.tink.HybridDecrypt;
+import com.google.crypto.tink.HybridEncrypt;
+import com.google.crypto.tink.InsecureSecretKeyAccess;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.TinkJsonProtoKeysetFormat;
+import com.google.crypto.tink.hybrid.HybridConfig;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Base64;
+
+// HybridReplaceTest
+class App {
+    public static void main(String[] args) throws Exception {
+        /*
+
+        Generated public/private keypairs for Alice and Bob using the
+        following tinkey commands:
+
+        ./tinkey create-keyset \
+        --key-template DHKEM_X25519_HKDF_SHA256_HKDF_SHA256_AES_256_GCM \
+        --out-format JSON --out alice_private_keyset.json
+
+        ./tinkey create-keyset \
+        --key-template DHKEM_X25519_HKDF_SHA256_HKDF_SHA256_AES_256_GCM \
+        --out-format JSON --out bob_private_keyset.json
+
+        ./tinkey create-public-keyset --in alice_private_keyset.json \
+        --in-format JSON --out-format JSON --out alice_public_keyset.json
+
+        ./tinkey create-public-keyset --in bob_private_keyset.json \
+        --in-format JSON --out-format JSON --out bob_public_keyset.json
+        */
+
+        HybridConfig.register();
+
+        // Generate ECC key pair for Alice
+        var alice = new HybridSimple(
+                getKeysetHandle("/home/josh/alice_private_keyset.json"),
+                getKeysetHandle("/home/josh/alice_public_keyset.json")
+
+        );
+
+        KeysetHandle alicePublicKey = alice.getPublicKey();
+
+        // Generate ECC key pair for Bob
+        var bob = new HybridSimple(
+                getKeysetHandle("/home/josh/bob_private_keyset.json"),
+                getKeysetHandle("/home/josh/bob_public_keyset.json")
+
+        );
+
+        KeysetHandle bobPublicKey = bob.getPublicKey();
+
+        // This keypair generation shoud be reperformed every so often in order to
+        // obtain a new shared secret to avoid a long lived shared secret.
+
+        // Alice encrypts a message to send to Bob
+        String plaintext = "Hello, Bob!";
+        String metadata = "Info about the message";

```

Comment:
Ditto. See previous comments on this. Maybe use something different for this example.

Reply:
See [above](https://github.com/OWASP/CheatSheetSeries/pull/1146#discussion_r1241143817).